### DISCUSSION TITLE
Adding structure to top-level AsyncGenerator variables?

### DISCUSSION THREAD

**Discussion by gouldingken** [observablehq/framework#885] [Q&A] [answered]
I have a number of input variables that repeat across categories. I was trying to group those into JS objects to allow them to be passed more easily to a function. 

```js
const stationOccupancyV = view(Inputs.range([0,100]))
const station = {
    "stationOccupancy": stationOccupancyV,
}
```

```js
stationOccupancyV
```

yields: `50`

```js
station.stationOccupancy
```

yields: 
```
AsyncGenerator {
  <prototype>: AsyncGenerator {}
}
```

I could use a prefix to get a unique name for each top level variable, but that would result in repetitive code as things become more complex. Is there a more elegant way to work with these async generators within a data structure that retains the behavior of the top-level variables?

**Comment by Fil** [comment, marked as answer]
A way could be Inputs.form which allows to combine several inputs into one: https://github.com/observablehq/inputs#inputsforminputs-options

**Reply by gouldingken** [reply]
That's a neat solution - thanks!
